app/lib/uniprot_api.js

```diff
@@ -7,69 +7,42 @@ const NCBI = require('./ncbi_api');
 // ex: For "Vulpes vulpes (red fox)", capture "Vulpes vulpes"
 const TAXON_EXTRACTING_REGEX = /^(\b[\w. ]+\b) *(?:\(.*)?$/;
 const QUERY_RESULT_LIMIT = 10;
-const BASE_URL = 'http://www.uniprot.org/uniprot/?format=json';
+const BASE_URL = 'http://www.ebi.ac.uk/proteins/api/proteins/';
 
 /**
- * Gets a uniprot gene by name.
+ * Gets a uniprot locus by name.
  * These can look like A2BC19, P12345, or A0A022YWF9 for Uniprot.
- * Promise resolves with the single matching gene,
- * or rejects with error for several or zero matching genes.
+ * Promise resolves with the single matching locus,
+ * or rejects with error with no matching locus.
  *
  * @param name
  * @returns {Promise}
  */
 function getLocusByName(name) {
-	let requestUrl = `${BASE_URL}&limit=2&query=accession:${name}`;
+	let requestUrl = `${BASE_URL}${name}.json`;
 	return new Promise((resolve, reject) => {
 		request.get(requestUrl, (error, response, bodyJson) => {
 			if (error) {
 				reject(new Error(error));
-			} else if (!bodyJson) {
+			} else if (response.statusCode != 200) {
 				reject(new Error(`No Locus found for name ${name}`));
 			} else {
 				let body = JSON.parse(bodyJson);
-				if (body.length > 1) {
-					reject(new Error('Given ID matches multiple loci'));
-				} else {
-					let taxonName = body[0].organism.match(TAXON_EXTRACTING_REGEX)[1];
-
-					NCBI.getTaxonByScientificName(taxonName).then(taxonInfo => {
-						resolve({
-							source: 'Uniprot',
-							locus_name: body[0].id,
-							taxon_name: taxonName,
-							taxon_id: taxonInfo.taxId
-						});
+				let taxonName = body.organism.names[0].value;
```

Comment:
Should implement search for scientific type field in body.organism.names unless we get confirmation that the first one in the array is of type scientific.

let taxon = body && body.organism && body.organism.names ? body.organism.names : undefined;
let taxonName = undefined;
if(taxon) {
    for(let item in taxon) {
        if(item.type && item.type == 'scientific') {
            taxonName = item.value;
        }
    }
}

if(taxonName) { ...